src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsExtensions.cs

```diff
@@ -212,45 +238,75 @@ public static IResourceBuilder<AzureEventHubsResource> RunAsEmulator(this IResou
             {
                 var configFileMount = emulatorResource.Annotations.OfType<ContainerMountAnnotation>().Single(v => v.Target == AzureEventHubsEmulatorResource.EmulatorConfigJsonPath);
 
-                using var stream = new FileStream(configFileMount.Source!, FileMode.Create);
-                using var writer = new Utf8JsonWriter(stream);
-
-                writer.WriteStartObject();                      // {
-                writer.WriteStartObject("UserConfig");          //   "UserConfig": {
-                writer.WriteStartArray("NamespaceConfig");      //     "NamespaceConfig": [
-                writer.WriteStartObject();                      //       {
-                writer.WriteString("Type", "EventHub");         //         "Type": "EventHub",
-
-                // This name is currently required by the emulator
-                writer.WriteString("Name", "emulatorNs1");      //         "Name": "emulatorNs1"
-                writer.WriteStartArray("Entities");             //         "Entities": [
-
-                foreach (var hub in emulatorResource.Hubs)
+                // If there is a custom mount for EmulatorConfigJsonPath we don't need to create the Config.json file.
+                if (configFileMount != defaultConfigFileMount)
                 {
-                    // The default consumer group ('$default') is automatically created
-
-                    writer.WriteStartObject();                  //           {
-                    writer.WriteString("Name", hub);            //             "Name": "hub",
-                    writer.WriteString("PartitionCount", "2");  //             "PartitionCount": "2",
-                    writer.WriteStartArray("ConsumerGroups");   //             "ConsumerGroups": [
-                    writer.WriteEndArray();                     //             ]
-                    writer.WriteEndObject();                    //           }
+                    continue;
                 }
 
-                writer.WriteEndArray();                         //         ] (/Entities)
-                writer.WriteEndObject();                        //       }
-                writer.WriteEndArray();                         //     ], (/NamespaceConfig)
-                writer.WriteStartObject("LoggingConfig");       //     "LoggingConfig": {
-                writer.WriteString("Type", "File");             //       "Type": "File"
-                writer.WriteEndObject();                        //     } (/LoggingConfig)
+                using (var stream = new FileStream(configFileMount.Source!, FileMode.Create))
+                {
+                    using var writer = new Utf8JsonWriter(stream, new JsonWriterOptions { Indented = true });
+
+                    if (!OperatingSystem.IsWindows())
+                    {
+                        File.SetUnixFileMode(configHostFile,
```

Comment:
We should be using the correct UnixFileMode when we create the file. This can be done by setting them on `new FileStreamOptions` and passing that into `new FileStream`.

This would fix the weirdness with using `configHostFile` here, but we are working on `configFileMount.Source`.